src/d3a/setup/two_sided_market/infinite_bus.py

```diff
@@ -0,0 +1,57 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from d3a.models.appliance.switchable import SwitchableAppliance
+from d3a.models.area import Area
+from d3a.models.strategy.load_hours import LoadHoursStrategy
+from d3a.models.strategy.infinite_bus import InfiniteBusStrategy
+from d3a.models.appliance.simple import SimpleAppliance
+from d3a.models.strategy.pv import PVStrategy
+from d3a.models.appliance.pv import PVAppliance
+from d3a.models.const import ConstSettings
+
+
+def get_setup(config):
+    ConstSettings.IAASettings.MARKET_TYPE = 2
+    config.update_config_parameters(transfer_fee_pct=0)
+    area = Area(
+        'Grid',
+        [
+            Area(
+                'House 1',
+                [
+                    Area('H1 General Load', strategy=LoadHoursStrategy(avg_power_W=200,
+                                                                       hrs_of_day=list(
+                                                                           range(0, 23)),
+                                                                       final_buying_rate=25),
+                         appliance=SwitchableAppliance()),
+                    Area('H1 PV', strategy=PVStrategy(panel_count=4, risk=0,
+                                                      final_selling_rate=23,
+                                                      initial_selling_rate=23,
+                                                      initial_rate_option=3),
+                         appliance=PVAppliance()),
+                ]
+            ),
+            Area('Infinite Bus', strategy=InfiniteBusStrategy(energy_buy_rate=24,
+                                                              energy_sell_rate=25),
+                 appliance=SimpleAppliance()
+                 ),
+
+        ],
+        config=config
+    )
+    return area
```

Comment:
I tested this setup file locally. What I see is Infinite bus is buying from PV and selling to load rather than allowing PV & Load to first trade mutually and then absorb the residual supply/demand. In my opinion, there should be a delay in InfiniteBus to post offers/bid so that it can only absorb residual supply/demand.

Reply:
You are right. The load should prefer the energy from the PV. But i dont know, whether implementing a delay would be a good solution. It should definitely be accessible for the user. I suggest for now to use the pricing configuration to achieve this. OK